.envrc

```diff
@@ -0,0 +1,11 @@
+if ! has nix_direnv_version || ! nix_direnv_version 2.2.1; then
```

Comment:
Is there any way to keep all these files out of the root folder? Theres already too much stuff.

Reply:
`.envrc` has probably have to stay, but all the rest can fit into `nix/` if necessary.

Reply:
`rust-toolchain.toml` will [have to stay](https://rust-lang.github.io/rustup/overrides.html#the-toolchain-file), too, because that's where rustup expects to find it. And moving the flake files to a subdir is possible but is somewhat non-conventional and will hinder the discoverability.